client/scss/components/_userbar.scss

```diff
@@ -213,53 +216,64 @@ $positions: (
   }
 }
 
-.w-dialog--userbar {
-  // Display off to the side of the page rather than in the middle.
-  inset-inline-start: auto;
-  font-family: theme('fontFamily.sans');
-  padding-inline-end: 2rem;
-  z-index: $userbar-z-index;
+@each $direction in $directions {
+  .w-dialog--userbar-#{$direction} {
+    font-family: theme('fontFamily.sans');
+    z-index: $userbar-z-index;
+
+    @if $direction == 'left' {
+      inset-inline-end: auto;
+      padding-inline-start: 2rem;
+    } @else {
+      inset-inline-start: auto;
+      padding-inline-end: 2rem;
+    }
```

Comment:
We try not to use too much of Sass-specific features ([per our guidelines](https://docs.wagtail.org/en/stable/contributing/ui_guidelines.html#sass-usage)). Also, we already have the directional styles for the userbar itself, so it'd be great if we could reuse that mechanism instead of redefining another of these conditional blocks:

https://github.com/wagtail/wagtail/blob/2e892b331cd1ac6afff8e08eb38a5571cc7dfde6/client/scss/components/_userbar.scss#L292-L330